### DISCUSSION TITLE
Concurrency in iteration block

### DISCUSSION THREAD

**Discussion by algorithmx** [langgenius/dify#5693] [Suggestion]
Using dify.ai web client, I have built a loop in a workflow to get answer to a series of 5 questions from LLM API. In most cases, these questions can be answered independently. So I want to  make the iterative execution to be concurrent. It seems that there is no way of achieving this from the web client. Do I need to modify the source code, or I wait for a relevant update?

**Comment by algorithmx** [comment]
still watching on this

**Comment by CLRafaelR** [comment]
I'm also curious about how we can make Iteration blocks work concurrently (or asynchronously). I think functions defined in the following two scripts should be implemented using the async method, but I don't know how personally... 

- [dify/api/core/workflow/nodes/iteration/iteration_node.py at main · langgenius/dify](https://github.com/langgenius/dify/blob/main/api/core/workflow/nodes/iteration/iteration_node.py)
- [dify/api/core/workflow/nodes/iteration/entities.py at main · langgenius/dify](https://github.com/langgenius/dify/blob/main/api/core/workflow/nodes/iteration/entities.py)

**Comment by tobegit3hub** [comment]
This has been implement and released in 0.8.0-beta1 https://github.com/langgenius/dify/discussions/7985 .
